tests/testthat/test-as_flex_table.R

```diff
@@ -0,0 +1,406 @@
+  # return_calls argument does not produce warnings
+  expect_silent(my_tbl_summary |> as_flex_table(return_calls = TRUE))
+
+  # include argument does not produce warnings
+  expect_silent(my_tbl_summary |> as_flex_table(include = tibble))
+
+  # correct elements are returned
+  expect_equal(
+    names(ft_tbl_summary),
+    c("header", "body", "footer", "col_keys", "caption", "blanks", "properties")
+  )
+})
+
+test_that("as_flex_table works with tbl_survfit", {
+  skip_if_not(is_pkg_installed("survival", reference_pkg = "gtsummary"))
+  fit1 <- survival::survfit(survival::Surv(ttdeath, death) ~ trt, trial)
+  tbl <- tbl_survfit(fit1, times = c(12, 24), label_header = "{time} Months")
+
+  expect_silent(ft_tbl <- tbl |> as_flex_table())
+
+  # check for correct table contents
+  expect_equal(
+    tbl |> as_tibble(col_labels = FALSE),
+    ft_tbl$body$dataset |> as_tibble(),
+    ignore_attr = "names"
+  )
+})
+
+test_that("as_flex_table works with tbl_merge", {
+  skip_if_not(is_pkg_installed("survival", reference_pkg = "gtsummary"))
+
+  t1 <- glm(response ~ trt + grade + age, trial, family = binomial) |>
+    tbl_regression(exponentiate = TRUE)
+  t2 <- survival::coxph(survival::Surv(ttdeath, death) ~ trt + grade + age, trial) |>
+    tbl_regression(exponentiate = TRUE)
+
+  tbl <- tbl_merge(
+    tbls = list(t1, t2),
+    tab_spanner = c("**Tumor Response**", "**Time to Death**")
+  )
+  ft_tbl <- tbl |> as_flex_table()
+
+  # header labels span correct columns
+  expect_equal(
+    ft_tbl$header$spans$rows[1, ],
+    c(1, 3, 0, 0, 3, 0, 0)
+  )
+  expect_equal(
+    ft_tbl$header$spans$rows[2, ],
+    rep(1, 7)
+  )
+
+  # spanning header labels are correct
+  expect_equal(
+    ft_tbl$header$dataset[1, ] |> unlist(use.names = FALSE),
+    c(" ", rep("**Tumor Response**", 3), rep("**Time to Death**", 3))
+  )
+})
+
+test_that("as_flex_table passes table contents correctly", {
+  # tbl_summary
+  vis_cols <- my_tbl_summary$table_styling$header |>
+    dplyr::filter(!hide) |>
+    dplyr::pull(column)
+
+  expect_equal(
+    my_tbl_summary$table_body[vis_cols],
+    ft_tbl_summary$body$dataset |> as_tibble()
+  )
+
+  # tbl_cross
+  my_tbl_cross <- tbl_cross(trial, trt, grade)
+  expect_silent(ft_tbl_cross <- my_tbl_cross |> as_flex_table())
+
+  expect_equal(
+    my_tbl_cross |> as_tibble(col_labels = FALSE),
+    ft_tbl_cross$body$dataset |> as_tibble()
+  )
+
+  # tbl_regression
+  expect_equal(
+    my_tbl_regression |> as_tibble(col_labels = FALSE),
+    ft_tbl_regression$body$dataset |> as_tibble(),
+    ignore_attr = c("class", "names")
+  )
+
+  # tbl_uvregression
+  my_tbl_uvregression <- trial |> tbl_uvregression(method = lm, y = age)
+  expect_silent(ft_tbl_uvregression <- my_tbl_uvregression |> as_flex_table())
+
+  expect_equal(
+    my_tbl_uvregression |> as_tibble(col_labels = FALSE),
+    ft_tbl_uvregression$body$dataset |> as_tibble(),
+    ignore_attr = c("class", "names")
+  )
+
+  # spanning_tbl
+  expect_equal(
```

Comment:
What are we testing with this one? Once we convert to a tibble, the spanning headers are gone?

Reply:
You're correct - I have removed this.